### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This is essentially an implementation of [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin) but targeting the legacy CyHy environment. The self-management policies are lifted directly from the `users` account configuration in [cisagov/cool-accounts](https://github.com/cisagov/cool-accounts).

## 💭 Motivation and context ##

Current user creation is spread out across multiple projects ([cisagov/findings-data-s3-users](https://github.com/cisagov/findings-data-s3-users) and [cisagov/bod-18-01-cloudwatch-users](https://github.com/cisagov/bod-18-01-cloudwatch-users)) and I am planning on making a new project to provide access to manage the users for the bastion instance. In addition to making it difficult to track non-admin users it makes it impossible for the same user to be in multiple permission groups. This has not been an issue in the past because of the low number of IAM accounts in the legacy CyHy environment, but it could be a lurking problem.

## ⚠ Implementation Notes ##

I will need to make changes to [cisagov/bod-18-01-cloudwatch-users](https://github.com/cisagov/bod-18-01-cloudwatch-users) and [cisagov/findings-data-s3-users](https://github.com/cisagov/findings-data-s3-users) to remove the user creation elements and consolidate all accounts on the user policies added in this repository.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -4,30 +4,55 @@
...
-  description = "The ID of the AWS account that owns the Example AMI, or \"self\" if the AMI is owned by the same account as the provisioner."
-  default     = "self"
 }
 
-variable "aws_availability_zone" {
+variable "non_self_admin_users" {
+  default     = []
+  description = "A list containing the usernames of non-admin users that are not allowed to administer their own accounts.  Example: [ \"service-account1\", \"service-account2\", \"service-account3\" ]"
+  type        = list(string)
+}
```
Comment (jsf9k) [inline variables.tf]: Any thoughts on adding this capability as another attribute in the `object` associated with each `username` in `var.users`?  That seems cleaner and less confusing, although it moves the complication to the Terraform code.

File: variables.tf [hunk 2]
```diff
@@ -4,30 +4,55 @@
 # You must provide a value for each of these parameters.
 # ------------------------------------------------------------------------------
 
-variable "subnet_id" {
-  type        = string
-  description = "The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0)."
+variable "users" {
+  description = "A map whose keys are the usernames of each non-admin user and whose values are a map containing supported user attributes.  The only currently-supported attribute is \"require_mfa\" (boolean).  Example: { \"firstname1.lastname1\" = { \"require_mfa\" = false }, \"firstname2.lastname2\" = { \"require_mfa\" = true }, \"firstname3.lastname3\" = { \"require_mfa\" = false } }"
+  type        = map(map(string))
```
Comment (jsf9k) [inline variables.tf]: This variable is currently of type `map(map(bool))` and would become more complicated if [this suggestion](https://github.com/cisagov/cyhy-users-non-admin/pull/5#discussion_r819646945) were implemented.  I think it makes sense to define it more precisely:
```suggestion
variable "users" {
  description = "A map whose keys are the usernames of each non-admin user and whose values are an object containing supported user attributes.  The only currently-supported attribute is \"require_mfa\" (boolean).  Example: { \"firstname1.lastname1\" = { \"require_mfa\" = false }, \"firstname2.lastname2\" = { \"require_mfa\" = true }, \"firstname3.lastname3\" = { \"require_mfa\" = false } }"
  type        = map(object({require_mfa = bool}))
```

**Review by jsf9k — CHANGES_REQUESTED** [review]
I have a requested change (which should be made to [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin) as well) and a question.

File: variables.tf (hunk 1 above)
Comment (mcdonnnj) [reply variables.tf]: I would be ok with that if we wanted to go that route. Something like a `bool` attribute called `self_admin` or something along those lines? What do you think @dav3r?

Comment (dav3r) [reply variables.tf]: Sure, that suggestion makes sense to me- I say go for it.

Comment (jsf9k) [reply variables.tf]: Assuming it works out, we should make the same changes to [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin).

File: variables.tf (hunk 2 above)
Comment (mcdonnnj) [reply variables.tf]: Please see [5c7c918](https://github.com/cisagov/cyhy-users-non-admin/pull/5/commits/5c7c918fd83aa08c8ca748cecbb16d997d0eecc2).

File: variables.tf (hunk 1 above)
Comment (mcdonnnj) [reply variables.tf]: With the approval of https://github.com/cisagov/cool-users-non-admin/pull/34 I have made similar changes in [eff3553](https://github.com/cisagov/cyhy-users-non-admin/pull/5/commits/eff35534dc9004ca10d45f7de26f37ada34258eb).

**Review by jsf9k — APPROVED** [review]
Thanks for making those changes, both here and in the [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin) repo!

**Review by dav3r — APPROVED** [review]
👍 👍  What @jsf9k said!  💪 💼

**Pull request merged**